java/ql/lib/ext/io.jsonwebtoken.model.yml

```diff
@@ -0,0 +1,15 @@
+extensions:
+  - addsTo:
+      pack: codeql/java-all
+      extensible: summaryModel
+    data:
+      - ["io.jsonwebtoken", "JwsHeader", True, "getAlgorithm", "", "", "Argument[this]", "ReturnValue", "taint", "manual"]
+      - ["io.jsonwebtoken", "JwsHeader", True, "setAlgorithm", "", "", "Argument[0]", "Argument[this]", "taint", "manual"]
+      - ["io.jsonwebtoken", "JwsHeader", True, "getKeyId", "", "", "Argument[this]", "ReturnValue", "taint", "manual"]
+      - ["io.jsonwebtoken", "JwsHeader", True, "setKeyId", "", "", "Argument[0]", "Argument[this]", "taint", "manual"]
```

Comment:
We're risking having spurious flow when tainting the header through the algorithm and getting the key ID, or vice-versa, e.g.

```java
header.setAlgorithm(tainted());
sink(header.getKeyId());
```

To solve that, we'd need to use synthetic fields:

```suggestion
      - ["io.jsonwebtoken", "JwsHeader", True, "getAlgorithm", "", "", "Argument[this].SyntheticField[io.jsonwebtoken.JwsHeader.algorithm]", "ReturnValue", "taint", "manual"]
      - ["io.jsonwebtoken", "JwsHeader", True, "setAlgorithm", "", "", "Argument[0]", "Argument[this].SyntheticField[io.jsonwebtoken.JwsHeader.algorithm]", "taint", "manual"]
      - ["io.jsonwebtoken", "JwsHeader", True, "getKeyId", "", "", "Argument[this].SyntheticField[io.jsonwebtoken.JwsHeader.keyId]", "ReturnValue", "taint", "manual"]
      - ["io.jsonwebtoken", "JwsHeader", True, "setKeyId", "", "", "Argument[0]", "Argument[this].SyntheticField[io.jsonwebtoken.JwsHeader.keyId]", "taint", "manual"]
```

But then you'll need a `TaintInheritingContent` to establish that a generally-tainted `JwsHeader` (like one that is a source) also taints its synthetic fields, like the one we have for `Intent.extras`:

https://github.com/github/codeql/blob/feb31612f5c4bfece15d233edcebc2069c3b50b5/java/ql/lib/semmle/code/java/frameworks/android/Intent.qll#L125-L129

I'll leave the final call to you. If you don't observe FPs caused by this spurious flow in real world applications, we can leave it as it is if you want.